test/e2e_node/util.go

```diff
@@ -397,8 +397,14 @@ func getCRIClient() (internalapi.RuntimeService, internalapi.ImageManagerService
 }
 
 // TODO: Find a uniform way to deal with systemctl/initctl/service operations. #34494
-func findRunningKubletServiceName() string {
-	stdout, err := exec.Command("sudo", "systemctl", "list-units", "*kubelet*", "--state=running").CombinedOutput()
+func findKubletServiceName(running bool) string {
```

Comment:
Is there a strict necessity to support this `bool`?

When running tests locally, you often end up with a lot of killed `*kubelet*` units, and this will break that behavior. I think the same might apply when reusing gce instances that tests run on. Not 100% sure tho., since I don't have all that in my head atm. :sweat_smile:

Reply:
If we should support it (and I guess we can), we would have to cache/save the correct unit somewhere, to make sure we only restart the correct one.

Reply:
this is not a public method, so i am not worried :)

Reply:
> When running tests locally, you often end up with a lot of killed *kubelet* units, and this will break that behavior.

I agree with this concern...

```
ehashman@fedora:~$ sudo systemctl list-units '*kubelet*' | wc -l
92
ehashman@fedora:~$ sudo systemctl list-units '*kubelet*' --state=running 
  UNIT LOAD ACTIVE SUB DESCRIPTION
0 loaded units listed.
ehashman@fedora:~$ sudo systemctl list-units '*kubelet*' | head
  UNIT                            LOAD   ACTIVE SUB    DESCRIPTION
● kubelet-20210723T100254.service loaded failed failed /home/ehashman/src/k8s/_output/local/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ehashman/src/k8s/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --network-plugin=kubenet --cni-bin-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/bin --cni-conf-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/net.d --cni-cache-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/cache --hostname-override fedora --container-runtime docker --container-runtime-endpoint unix:///var/run/dockershim.sock --config /home/ehashman/src/k8s/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local --container-runtime=docker --network-plugin= --cni-bin-dir=
● kubelet-20210723T101707.service loaded failed failed /home/ehashman/src/k8s/_output/local/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ehashman/src/k8s/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --feature-gates DynamicKubeletConfig=true,LocalStorageCapacityIsolation=true --dynamic-config-dir /home/ehashman/src/k8s/_output/local/go/bin/dynamic-kubelet-config --network-plugin=kubenet --cni-bin-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/bin --cni-conf-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/net.d --cni-cache-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/cache --hostname-override fedora --container-runtime docker --container-runtime-endpoint unix:///var/run/dockershim.sock --config /home/ehashman/src/k8s/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local --container-runtime=docker --network-plugin= --cni-bin-dir= --cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false
● kubelet-20210723T102028.service loaded failed failed /home/ehashman/src/k8s/_output/local/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ehashman/src/k8s/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --feature-gates DynamicKubeletConfig=true,LocalStorageCapacityIsolation=true --dynamic-config-dir /home/ehashman/src/k8s/_output/local/go/bin/dynamic-kubelet-config --network-plugin=kubenet --cni-bin-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/bin --cni-conf-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/net.d --cni-cache-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/cache --hostname-override fedora --container-runtime docker --container-runtime-endpoint unix:///var/run/dockershim.sock --config /home/ehashman/src/k8s/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local --container-runtime=docker --network-plugin= --cni-bin-dir= --cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false
● kubelet-20210723T102356.service loaded failed failed /home/ehashman/src/k8s/_output/local/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ehashman/src/k8s/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --feature-gates DynamicKubeletConfig=true,LocalStorageCapacityIsolation=true --dynamic-config-dir /home/ehashman/src/k8s/_output/local/go/bin/dynamic-kubelet-config --network-plugin=kubenet --cni-bin-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/bin --cni-conf-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/net.d --cni-cache-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/cache --hostname-override fedora --container-runtime docker --container-runtime-endpoint unix:///var/run/dockershim.sock --config /home/ehashman/src/k8s/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local --container-runtime=docker --network-plugin= --cni-bin-dir= --fail-swap-on=false
  kubelet-20210723T102644.service loaded active exited /home/ehashman/src/k8s/_output/local/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ehashman/src/k8s/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --feature-gates DynamicKubeletConfig=true,LocalStorageCapacityIsolation=true --dynamic-config-dir /home/ehashman/src/k8s/_output/local/go/bin/dynamic-kubelet-config --network-plugin=kubenet --cni-bin-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/bin --cni-conf-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/net.d --cni-cache-dir /home/ehashman/src/k8s/_output/local/go/bin/cni/cache --hostname-override fedora --container-runtime docker --container-runtime-endpoint unix:///var/run/dockershim.sock --config /home/ehashman/src/k8s/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local --container-runtime=docker --network-plugin= --cni-bin-dir= --cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --fail-swap-on=false
...
```

Reply:
> @dims: this is not a public method, so i am not worried :)

Sorry, maybe I am missing some context, but is there a reason that is an important aspect here?

> @ehashman: I agree with this concern...

Thanks for looking @ehashman!

This would "essentially" mean that I/the developer will have to reboot my/their computer each time I/they want to run the test, or spin up a new gce instance for each time, and imo. that is not the best developer experience :/

Reply:
I'm okay with addressing this in a follow-up fwiw.